### PR TITLE
Add more tools to CI and docs

### LINKED ISSUE #187: auto-run documented tool programs
In a CI workflow, run each documented tool program, like we currently run org.unicode.text.UCD.Main. Each tool should run successfully (not crash, not fail). For this issue, ignore the output.
- Look through the https://github.com/unicode-org/unicodetools/tree/main/docs pages (excluding "help")
- For each tool documented there,
  - Add it to the list below
  - Add it to CI

List of tools:
- [x] AacOrder.java, see https://github.com/unicode-org/unicodetools/blob/main/docs/emoji/aac.md#aacorderjava
- [ ] EmojiFrequency.java, see https://github.com/unicode-org/unicodetools/blob/main/docs/emoji/emoji-frequency.md#processing
  - May need a dummy .tsv file
- [x] org.unicode.text.UCA.Main, see https://github.com/unicode-org/unicodetools/blob/main/docs/uca/index.md#tools--tests
- [x] GenerateIdna.java, see https://github.com/unicode-org/unicodetools/blob/main/docs/idna.md
- [x] GenerateIdnaTest.java, see https://github.com/unicode-org/unicodetools/blob/main/docs/idna.md
- [x] GenerateEnums.java, see https://github.com/unicode-org/unicodetools/blob/main/docs/newunicodeproperties.md#run-generateenumsjava
  - Would be useful to fail if the generated output differs from the repo files.
- [x] GenerateConfusables, see https://github.com/unicode-org/unicodetools/blob/main/docs/security.md#generating
- [x] GenerateUnihanCollators, see https://github.com/unicode-org/unicodetools/blob/main/docs/unihan.md#run-generateunihancollators
- [x] CheckProperties, see https://github.com/unicode-org/unicodetools/blob/main/docs/newunicodeproperties.md#checking-other-properties

Not a tool:
- ToolUnicodePropertySource.java, see https://github.com/unicode-org/unicodetools/blob/main/docs/changing-ucd-properties.md#toolunicodepropertysourcejava

Skip:
- GenerateUnihanCollatorFiles, see https://github.com/unicode-org/unicodetools/blob/main/docs/unihan.md#run-generateunihancollatorfiles
- GenerateEmoji.java, see https://github.com/unicode-org/unicodetools/blob/main/docs/emoji/generate-emoji.md#running-the-tool
  - 2022-mar-22: out of scope because it needs access to another repo, see [discussion below](https://github.com/unicode-org/unicodetools/issues/187#issuecomment-1075566958)

### PR DESCRIPTION
**Pull request by echeran**
Contributes to #187

### CODE REVIEW

**Comment by markusicu** [comment]
@echeran Good time to make progress?
Maybe check in one PR for some tools and then create one or more additional ones?

**Comment by echeran** [comment]
@markusicu Good idea, I changed this PR from "draft" to "ready for review".

File: .github/workflows/cli-build-instructions.yml [hunk 1]
```diff
@@ -41,15 +41,59 @@ jobs:
...
+      #     # Confirm that the directory for the bin files exists
+      #     compgen -G "Generated/BIN"
+      #     # "Delete all the bin files" as per instructions
+      #     rm -rf Generated/BIN/*
+      #     # run GenerateIdna
+      #     MAVEN_OPTS="-ea" mvn -s .github/workflows/mvn-settings.xml exec:java -Dexec.mainClass="org.unicode.idna.GenerateIdna" -Dexec.args="" -pl unicodetools  -DCLDR_DIR=$(cd ../cldr ; pwd)  -DUNICODETOOLS_GEN_DIR=$(cd Generated; pwd)  -DUNICODETOOLS_REPO_DIR=$(pwd)  -DUVERSION=15.0.0
+      #   env:
+      #     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
   out-of-source-build:
```
Comment (markusicu) [inline .github/workflows/cli-build-instructions.yml:97]: optional: My preference is to only test with the out-of-source builds, which are a "stronger" type of test in a way. Duplicate tests/tool runs add to the maintenance burden. It's ok to test/run one thing (or a small number).

File: docs/build.md [hunk 2]
```diff
@@ -149,12 +149,24 @@ Run the "Build and Test" command below to compile all of the source files. Some
 Common tasks for Unicode Tools are listed below with example CLI commands with example argument values that they need:
 
 - Make Unicode Files:
-  * Out-of-source build: `mvn -s .github/workflows/mvn-settings.xml exec:java -Dexec.mainClass="org.unicode.text.UCD.Main"  -Dexec.args="version 14.0.0 build MakeUnicodeFiles"  -pl unicodetools  -DCLDR_DIR=$(cd ../../../cldr/mine/src ; pwd)  -DUNICODETOOLS_GEN_DIR=$(cd ../Generated ; pwd)  -DUNICODETOOLS_REPO_DIR=$(pwd)  -DUVERSION=14.0.0`
-  * In-source build: `MAVEN_OPTS="-ea" mvn exec:java -Dexec.mainClass="org.unicode.text.UCD.Main"  -Dexec.args="version 14.0.0 build MakeUnicodeFiles"  -pl unicodetools  -DCLDR_DIR=$(cd ../cldr ; pwd)  -DUNICODETOOLS_GEN_DIR=$(cd Generated; pwd)  -DUNICODETOOLS_REPO_DIR=$(pwd)  -DUVERSION=14.0.0`
+  * <details><summary>Out-of-source build:</summary>
```
Comment (markusicu) [inline docs/build.md:152]: Elango says that he wants to revisit these edits...

File: .github/workflows/cli-build-instructions.yml [hunk 3]
```diff
@@ -41,15 +41,59 @@ jobs:
...
+      # - name: Run command - IDNA
+      #   run: |
+      #     # Confirm that the directory for the bin files exists
+      #     compgen -G "Generated/BIN"
+      #     # "Delete all the bin files" as per instructions
+      #     rm -rf Generated/BIN/*
+      #     # run GenerateIdna
+      #     MAVEN_OPTS="-ea" mvn -s .github/workflows/mvn-settings.xml exec:java -Dexec.mainClass="org.unicode.idna.GenerateIdna" -Dexec.args="" -pl unicodetools  -DCLDR_DIR=$(cd ../cldr ; pwd)  -DUNICODETOOLS_GEN_DIR=$(cd Generated; pwd)  -DUNICODETOOLS_REPO_DIR=$(pwd)  -DUVERSION=15.0.0
+      #   env:
+      #     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (markusicu) [inline .github/workflows/cli-build-instructions.yml:95]: Please rebase and try this again. It should work after Steven's recent changes.

File: .github/workflows/cli-build-instructions.yml (hunk 1 above)
Comment (echeran) [reply .github/workflows/cli-build-instructions.yml:60]: Done. I updated the name of the CI job for in-source build to say "(Examples)", and did not add any new in-source versions of commands executing the tools listed in the issue.

File: .github/workflows/cli-build-instructions.yml (hunk 3 above)
Comment (echeran) [reply .github/workflows/cli-build-instructions.yml]: Done.

File: docs/build.md (hunk 2 above)
Comment (echeran) [reply docs/build.md]: Done.

File: .github/workflows/cli-build-instructions.yml [hunk 4]
```diff
@@ -9,8 +9,13 @@ on:
     - '*'
 
 jobs:
+
+  # Only run 1 or 2 commands for in-source build instructions, just to demonstrate
+  # what translating between an out-of-source build and an in-source build would
+  # would look like. Rely on out-of-source build steps as the canonical way to
+  # execute tools in CI and write corresponding documentation.
```
Comment (markusicu) [inline .github/workflows/cli-build-instructions.yml:16]:
```suggestion
  # Only run 1 or 2 commands for in-source build instructions, just to demonstrate
  # what translating between an out-of-source build and an in-source build would
  # look like. Rely on out-of-source build steps as the canonical way to
  # execute tools in CI and write corresponding documentation.
```

**Review by markusicu — APPROVED** [review]
lgtm tnx!

please apply one more typo fix and then squash-and-merge

**Review by markusicu — APPROVED** [review]
lgtm

**Pull request merged**